pgmpy/factors/JointProbabilityDistribution.py

```diff
@@ -252,16 +291,17 @@ def minimal_imap(self, order):
         """
         from pgmpy.models import BayesianModel
 
-        def combinations(u):
+        def get_subsets(u):
             for r in range(len(u) + 1):
                 for i in itertools.combinations(u, r):
                     yield i
 
         G = BayesianModel()
         for variable_index in range(len(order)):
-            u = order[:variable_index]
-            for subset in combinations(u):
-                if self.check_independence(order[variable_index], set(u)-set(subset), subset):
+            u = order[:variable_index+1]
```

Comment:
@ankurankan Problem is basically that how to get to the last element without getting out of index for `[order[variable_index]]` should I add an if condition , but I won't like it unless it is a last resort.